File: mobile/android/app/src/main/java/com/findly/android/ui/onboarding/OnboardingScreen.kt
package com.findly.android.ui.onboarding

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.findly.android.ui.designsystem.FindlyTheme
import com.findly.android.ui.designsystem.components.FindlyButton
import com.findly.android.ui.designsystem.components.FindlyButtonStyle
import com.findly.android.ui.designsystem.components.FindlyEmptyState
import com.findly.android.ui.designsystem.components.FindlyTextField
import com.findly.android.ui.designsystem.components.FindlyTopBar

/**
 * The 010-app-shell-and-screen-ux.md §2.2 Onboarding screen — one route, two variants, replacing
 * the two places this UI lived before this spec (iOS Home's `profileless`/`familyless` branches;
 * Android's `GroupsListScreen` `ProfileNeeded` state — both retired; the A21/I17 behaviors move
 * here unchanged, including the blank-name guard on all four bootstrap paths, which already lives
 * in [com.findly.android.ui.family.CreateFamilyStateHolder.validate] /
 * [com.findly.android.ui.invites.InvitesStateHolder.acceptInvite] /
 * [com.findly.android.ui.groups.CreateGroupStateHolder.validate] /
 * [com.findly.android.ui.groups.GroupJoinStateHolder.validate] and needs no duplicate here).
 *
 * Onboarding is a **root** (010 §2.2: "no back affordance, no drawer") — reached only via a full
 * stack reset (either the 010 §1.1 launch gate, or the 010 §2.1 dead-end routing rule), so there
 * is deliberately nothing behind it to go back to; [FindlyTopBar] renders no back control here
 * simply because [com.findly.android.ui.designsystem.components.LocalNavBackAction] is `null` at
 * this point in the stack, the same mechanism every other screen relies on (specs/003 §12.5) —
 * this screen does not special-case it.
 *
 * No dedicated `StateHolder`/`ViewModel`: this screen holds only the once-typed display-name text
 * (profile-less variant), carried into whichever bootstrap destination the user picks — the exact
 * "enter it once" behavior [com.findly.android.ui.groups.GroupsListScreen]'s retired `ProfileNeeded`
 * branch had.
 */
@Composable
fun OnboardingScreen(
    variant: OnboardingVariant,
    modifier: Modifier = Modifier,
    onCreateFamily: (displayName: String) -> Unit = {},
    onAcceptInvite: (displayName: String) -> Unit = {},
    onCreateGroup: (displayName: String) -> Unit = {},
    onJoinGroup: (displayName: String) -> Unit = {},
    onOpenGroups: () -> Unit = {},
    onOpenPrivacy: () -> Unit = {},
) {
    var displayName by remember { mutableStateOf("") }

    Column(modifier = modifier.fillMaxSize()) {
        FindlyTopBar(title = "Welcome to Findly")

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(FindlyTheme.spacing.md),
            verticalArrangement = Arrangement.spacedBy(FindlyTheme.spacing.sm),
        ) {
            when (variant) {
                OnboardingVariant.ProfileLess -> {
                    FindlyEmptyState(
                        title = "Welcome to Findly",
                        message = "Create or join a family, or start a temporary group — pick how you'd like to get started.",
                    )
                    FindlyTextField(value = displayName, onValueChange = { displayName = it }, label = "Your display name")

                    FindlyButton(text = "Create a family", onClick = { onCreateFamily(displayName) })
                    FindlyButton(
                        text = "I have an invite code",
                        onClick = { onAcceptInvite(displayName) },
                        style = FindlyButtonStyle.Secondary,
                    )
                    FindlyButton(
                        text = "Create a group",
                        onClick = { onCreateGroup(displayName) },
                        style = FindlyButtonStyle.Secondary,
                    )
                    FindlyButton(
                        text = "Join a group",
                        onClick = { onJoinGroup(displayName) },
                        style = FindlyButtonStyle.Secondary,
                    )
                }

                OnboardingVariant.FamilyLess -> {
                    FindlyEmptyState(
                        title = "No family yet",
                        message = "Create a family, join one with an invite code, or keep using groups.",
                    )
                    FindlyButton(text = "Create a family", onClick = { onCreateFamily(displayName) })
                    FindlyButton(
                        text = "I have an invite code",
                        onClick = { onAcceptInvite(displayName) },
                        style = FindlyButtonStyle.Secondary,
                    )
                    FindlyButton(text = "Groups", onClick = onOpenGroups, style = FindlyButtonStyle.Secondary)
                }
            }

            FindlyButton(text = "Privacy & data", onClick = onOpenPrivacy, style = FindlyButtonStyle.Secondary)
        }
    }
}

@Preview(name = "Onboarding — profile-less", showBackground = true)
@Composable
private fun OnboardingScreenProfileLessPreview() {
    FindlyTheme(darkTheme = false) {
        OnboardingScreen(variant = OnboardingVariant.ProfileLess)
    }
}

@Preview(name = "Onboarding — family-less", showBackground = true)
@Composable
private fun OnboardingScreenFamilyLessPreview() {
    FindlyTheme(darkTheme = true) {
        OnboardingScreen(variant = OnboardingVariant.FamilyLess)
    }
}

File: mobile/ios/FindlyKit/Sources/FindlyKit/Screens/Onboarding/OnboardingScreen.swift
import SwiftUI

/// specs/010-app-shell-and-screen-ux.md §2.2 — the single Onboarding screen, replacing the retired
/// Home hub's `.profileless`/`.familyless` branches (I17's behaviors move here unchanged, incl. the
/// A21 blank-name guard, which lives at each of the four bootstrap view models — see
/// `CreateFamilyViewModel`/`AcceptInviteViewModel`/`CreateGroupViewModel`/`GroupJoinViewModel`).
///
/// A root: no back affordance (nothing behind it worth going back to — every family screen would
/// fail the same way), no drawer (010 §2.2's own words). `RootView` never gives this screen a
/// `navBarBackAction`/drawer wiring, so there is nothing to omit here either.
///
/// Composes ONLY design-system components, exactly like every other screen (004 §2.4) — no network
/// call of its own; the four bootstrap paths are separate routes/view models this screen only
/// navigates to.
public struct OnboardingScreen: View {
    @Environment(\.theme) private var theme
    private let variant: OnboardingVariant
    // The display name typed once here is carried (still editable) into whichever bootstrap path
    // the user picks — the exact "remembered local state" pattern the retired Home hub used
    // (`pendingOnboardingDisplayName`, threaded by `RootView`), not owned by this screen's own
    // navigation callbacks.
    @State private var displayName: String
    private let onSelectCreateFamily: (String) -> Void
    private let onSelectAcceptInvite: (String) -> Void
    private let onSelectCreateGroup: (String) -> Void
    private let onSelectJoinGroup: (String) -> Void
    private let onSelectGroups: () -> Void
    private let onSelectPrivacySettings: () -> Void

    public init(
        variant: OnboardingVariant,
        prefillDisplayName: String = "",
        onSelectCreateFamily: @escaping (String) -> Void,
        onSelectAcceptInvite: @escaping (String) -> Void,
        onSelectCreateGroup: @escaping (String) -> Void = { _ in },
        onSelectJoinGroup: @escaping (String) -> Void = { _ in },
        onSelectGroups: @escaping () -> Void = {},
        onSelectPrivacySettings: @escaping () -> Void
    ) {
        self.variant = variant
        self._displayName = State(initialValue: prefillDisplayName)
        self.onSelectCreateFamily = onSelectCreateFamily
        self.onSelectAcceptInvite = onSelectAcceptInvite
        self.onSelectCreateGroup = onSelectCreateGroup
        self.onSelectJoinGroup = onSelectJoinGroup
        self.onSelectGroups = onSelectGroups
        self.onSelectPrivacySettings = onSelectPrivacySettings
    }

    public var body: some View {
        VStack(spacing: 0) {
            FindlyNavBar("Findly")
            content
        }
        .background(theme.colors.surfaceVariant)
    }

    @ViewBuilder
    private var content: some View {
        switch variant {
        case .profileLess:
            profileLessContent
        case .familyLess:
            familyLessContent
        }
    }

    /// 001 §1.5.3 — no `Users` profile row exists at all, so only the four bootstrap paths make
    /// sense; `GET /groups` would 404 too (§12.2 requires a profile), so Groups is deliberately
    /// NOT offered here.
    private var profileLessContent: some View {
        ScrollView {
            VStack(spacing: theme.spacing.md) {
                EmptyStateView(
                    title: "Welcome to Findly",
                    message: "Create or join a family, or start a temporary group — pick how you'd like to get started."
                )
                FindlyTextField("Your display name", text: $displayName, placeholder: "Eric")
                FindlyButton("Create a family") { onSelectCreateFamily(displayName) }
                FindlyButton("I have an invite code", style: .secondary) { onSelectAcceptInvite(displayName) }
                FindlyButton("Create a group", style: .secondary) { onSelectCreateGroup(displayName) }
                FindlyButton("Join a group", style: .secondary) { onSelectJoinGroup(displayName) }
                // 001 §13.2 / 008 §4.1 permits DELETE /users/me even with no profile — a brand-new
                // user who decides not to proceed with onboarding still needs a way out.
                FindlyButton("Privacy & data", style: .secondary) { onSelectPrivacySettings() }
            }
            .padding(theme.spacing.xl)
        }
    }

    /// 001 §1.5.4 — a profile exists (`familyId: null`); Groups (their one live feature) is
    /// unconditionally reachable, and no display-name field is needed (they already have one).
    private var familyLessContent: some View {
        ScrollView {
            VStack(spacing: theme.spacing.md) {
                EmptyStateView(
                    title: "No family yet",
                    message: "Create or join a family, or keep using your temporary group."
                )
                FindlyButton("Create a family") { onSelectCreateFamily(displayName) }
                FindlyButton("I have an invite code", style: .secondary) { onSelectAcceptInvite(displayName) }
                FindlyButton("Groups", style: .secondary) { onSelectGroups() }
                FindlyButton("Privacy & data", style: .secondary) { onSelectPrivacySettings() }
            }
            .padding(theme.spacing.xl)
        }
    }
}
